new-frontiers-14/frontier-station-14
# Fix artifact analyzers to consume power only during scanning

## About the PR



Currently artifact analyzers consume 12000W regardless of whether they are in use or not. In the YML it is commented that the intention is to only consume power during scans, but this is not the case because the APC load doesn't get updated. I think there was confusion about what apc.NeedsPower actually does, as it shouldn't affect the receiver's load (but rather, the receiver's requirement for power in the first place).

Technically wattage is set to 1 while not scanning for upstream APC bugs reasons, as @dvir001 mentioned earlier.

## Why / Balance


It would certainly reduce power / fuel requirements for science ships. However the same effect is achieved by unanchoring the analyzer. Currently it's just an extra step that feels lrp.

## How to test


Spawn a science ship, place artifact and scan / finish scan while monitoring power. Experiment with toggling power during scans as well to make sure pause / resume logic wasn't affected.

## Media


![2024-09-23-162234_586x750_scrot](https://github.com/user-attachments/assets/8c6a5aca-90d5-414e-9067-9bbf1c42d129)
![2024-09-23-162219_653x819_scrot](https://github.com/user-attachments/assets/49b66d6a-00fa-4204-8510-86d4622b70bd)

## Requirements


## Breaking changes


Not a breaking change, but something I'd like to do better before this is merged: when the analyzer begins scanning, the sudden 0->12kW power draw causes apc brownout. This
1) closes the analyze console UI. not the friendliest effect.
2) Emits a PowerChanged event, which then pauses the scan, then is issued again once the APC is ramped up to resume the scan. Pausing and resume the scan doesn't call  SetPowerSwitch(...) because it's actually only called when power is lost and restored, and only concerns the ActiveArtifactAnalyzerComponent. Anyway, it's kinda noisy and waste of cpu cycles.

ideas?

**Changelog**

:cl:
- fix: Artifact analyzers do not consume power when idle
